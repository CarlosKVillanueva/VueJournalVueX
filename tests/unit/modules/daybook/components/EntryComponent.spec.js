import { shallowMount } from "@vue/test-utils"
import EntryComponent from "@/modules/daybook/components/EntryComponent"
import { journalState } from "../../../mocks/test-journal-state"


describe( 'EntryComponent Suite Test', () => {

    const mockRouter = {
        push: jest.fn()
    }

    const wrapper = shallowMount( EntryComponent, {
       props: {
           entry: journalState.entries[ 0 ]
       },
        global: {
           mocks: {
               $router: mockRouter
           }
        }
    } )

    test('Should match Snapshot', () => {

        expect( wrapper.html() ).toMatchSnapshot()

    });

    test( 'Should redirect on entry-container click', () => {
        const entryContainer = wrapper.find( '.entry-container' )

        entryContainer.trigger( 'click' )

        expect( mockRouter.push).toHaveBeenCalledWith( {
            name: 'entry',
            params: {
                id: journalState.entries[ 0 ].id
            }
        })
    } );

    test('Test on computed props', () => {

        // expect( wrapper.vm.day ).toBe()
        expect( wrapper.vm.day ).toBe(13)
        expect( wrapper.vm.month ).toBe('Agosto')
        expect( wrapper.vm.yearDay ).toBe('2022, SÃ¡bado')


    });
} );