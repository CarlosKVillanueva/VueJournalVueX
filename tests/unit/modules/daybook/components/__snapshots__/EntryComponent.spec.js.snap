// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntryComponent Suite Test Should match Snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <div class="entry-title"><span class="text-success fs-5 fw-bold">13</span><span class="mx-1 fs-5">Agosto</span><span class="mx-2 fw-light">2022, SÃ¡bado</span></div>
  <div class="entry-description">Slam Dunk, el mejor anime deportivo de la historia!!!!</div>
</div>
`;
